include/dxc/Test/DxcTestUtils.h

```diff
@@ -175,10 +175,17 @@ std::string DisassembleProgram(dxc::DxcDllSupport &dllSupport, IDxcBlob *pProgra
 void SplitPassList(LPWSTR pPassesBuffer, std::vector<LPCWSTR> &passes);
 void MultiByteStringToBlob(dxc::DxcDllSupport &dllSupport, const std::string &val, UINT32 codePoint, _Outptr_ IDxcBlob **ppBlob);
 void MultiByteStringToBlob(dxc::DxcDllSupport &dllSupport, const std::string &val, UINT32 codePoint, _Outptr_ IDxcBlobEncoding **ppBlob);
-void ReplaceDisassemblyText(llvm::ArrayRef<LPCSTR> pLookFors, llvm::ArrayRef<LPCSTR> pReplacements,
-                            bool bRegex, std::string& disassembly);
-void ReplaceDisassemblyText(llvm::ArrayRef<LPCSTR> pLookFors, llvm::ArrayRef<LPCSTR> pReplacements,
-                           _Outptr_ IDxcBlob **pBlob, bool bRegex, std::string& disassembly, dxc::DxcDllSupport &dllSupport);
+void ReplaceDisassemblyText(llvm::ArrayRef<LPCSTR> pLookFors,
+                            llvm::ArrayRef<LPCSTR> pReplacements, bool bRegex,
+                            std::string &disassembly);
+void ReplaceDisassemblyText(llvm::ArrayRef<LPCSTR> pLookFors,
+                            llvm::ArrayRef<LPCSTR> pReplacements,
+                            _Outptr_ IDxcBlob **pBlob, bool bRegex,
+                            std::string &disassembly,
+                            dxc::DxcDllSupport &dllSupport);
+void ReplaceDisassemblyTextWithRegex(llvm::ArrayRef<LPCSTR> pLookFors,
+                                     llvm::ArrayRef<LPCSTR> pReplacements,
```

Comment:
I'm wondering if these functions that use `ArrayRef<LPCSTR>` could be changed to use `const std::vector<std::string>&` for consistency with the `ReplaceDisassemblyTextWithoutRegex` function, and to avoid llvm::ArrayRef dependency here.  I wonder how many other functions would have to be changed in individual test modules for this.  Perhaps in a follow-up change.

Reply:
Yes, this should be left for a follow-up change.